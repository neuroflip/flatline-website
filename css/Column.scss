#col1, #col2, #col3 {
    position: fixed;
    height: calc(100% - 85px);
    width: 90%;
    // max-width: 375px;
    overflow-y: scroll;
    margin: 10px auto;
    padding: 0 5px;
    border-radius: 21px;
    z-index: -10;

    a {
        text-decoration: none;
        -webkit-user-drag: none;
    }
    
    img {
        -webkit-user-drag: none;
        background-size: cover;
        -webkit-box-reflect: below 0px -webkit-gradient(linear, left top, left bottom, from(transparent), color-stop(0.95, transparent), to(#777));
        margin: 20px auto;
        display: block;
        max-width: 250px;
    }
    
    .textContainer {
        background-color: rgba(255,255,255,1.0);
        padding: 0 20px;
        border-radius: 15px;
        border: 4px solid cadetblue;
    }
    
    .textContainer:before {
        display: block;
        margin: 0 auto;
        width: 80%;
        max-width: 200px;
        color: white;
        position: relative;
        top: -27px;
        border-top-left-radius: 10px;
        border-top-right-radius: 10px;
        font-family: 'Lucida Sans', sans-serif;
        background-color: cadetblue;
        text-align: center;
        line-height: 21px;
        padding: 3px;
    }
    
    .coverTitle,
    .coverText {
        width: 100%;
        margin: 0px auto;
        color: rgba(0,0,0,1.0);
        background-color: rgba(255,255,255,1.0);
        font-family: 'Lucida Sans', sans-serif;
    }
    
    .coverTitle {
        font-size: 2em;
        margin-top: 20px;
    }
    
    .columnContainer {
        width: 95%;
        margin: 20px auto;
        background-color: #000;
        border: 1px solid #000;
        border-radius: 15px;
    }
}

#col1.animated,
#col2.animated,
#col3.animated {
    transition: 0.1s ease-in-out;
    pointer-events: none;
}

.col1::-webkit-scrollbar,
.col2::-webkit-scrollbar,
.col3::-webkit-scrollbar {
    display: none;
}

#col1 {
    transform: translate(0px, 0);
}

#col2 {
    transform: translate(700px, 0);
}

#col1.current,
#col2.current,
#col3.current {
    z-index: 10;
}


