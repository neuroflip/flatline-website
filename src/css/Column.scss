#col1, #col2, #col3 {
    position: fixed;
    height: 100%;
    width: 88%;
    
    overflow-y: scroll;
    margin: 0 auto;
    padding: 0 5px;
    border-radius: 21px;
    z-index: -10;
    opacity: 0.4;

    a {
        margin: 0 auto;
        display: block;
        text-decoration: none;
        -webkit-user-drag: none;
    }
    
    .image {
        width: 100%;
        height: 300px;
        background-size: cover;
        border-radius: 15px 15px 0 0;
    }
    
    .textContainer {
        background-color: rgba(255,255,255,1.0);
        padding: 0 30px 20px 20px;
        border-radius: 0 0 15px 15px;
        font-family: 'Victor Mono';
    }
    
    .textContainer:before {
        display: block;
        margin: 0 auto;
        width: 80%;
        max-width: 200px;
        color: white;
        position: relative;
        top: -27px;
        border-top-left-radius: 10px;
        border-top-right-radius: 10px;
        background-color: cadetblue;
        text-align: center;
        line-height: 21px;
        padding: 3px;
        font-size: 0.8em;
    }
    
    .coverTitle,
    .coverText {
        width: 100%;
        margin: 0px auto;
        color: rgba(0,0,0,1.0);
        background-color: rgba(255,255,255,1.0);
    }
    
    .coverTitle {
        font-size: 2em;
        margin: 40px 0;
        text-align: center;
    }

    .coverText {
        margin: 40px 10px;
    }

    .columnContainer {
        width: 95%;
        margin: 20px auto;
        box-shadow: 5px 5px rgba(80, 80, 80, 0.7);
        border-radius: 14px;
    }
}

#col1.animated,
#col2.animated,
#col3.animated {
    transition: 0.4s ease-in-out;
    pointer-events: none;
}

.col1::-webkit-scrollbar,
.col2::-webkit-scrollbar,
.col3::-webkit-scrollbar {
    display: none;
}

#col1 {
    transform: translate(0px, 0);
}

#col2 {
    transform: translate(700px, 0);
}

#col1.current,
#col2.current,
#col3.current {
    z-index: 10;
    opacity: 1.0;
}

@media only screen and (min-width : 500px) {
    #col1, #col2, #col3 {
        width: 40%;
    }
}
