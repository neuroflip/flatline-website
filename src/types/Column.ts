interface ColumnProps {
  id: string
  className: string
  style: React.CSSProperties
  children: any
}

export type { ColumnProps }
