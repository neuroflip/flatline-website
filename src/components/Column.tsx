import * as React from 'react'
import type { ColumnProps } from '../types/Column'

import '../../css/Column.scss'

function Column ({ id, className, style, children }: ColumnProps): React.JSX.Element {
  return (<div id={id} className={className} style={style} >
    {children}
  </div>)
}

export default Column
