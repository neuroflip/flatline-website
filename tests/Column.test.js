/**
 * @jest-environment jsdom
 */

import * as React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import { screen } from '@testing-library/dom'
import Column from '../src/components/Column'

let container = null

beforeEach(() => {
  container = window.document.createElement('div')
  container.id = 'root'
  window.document.body.appendChild(container)
})

afterEach(() => {
  window.document.body.removeChild(container)
})

it('Column renders as expected', () => {
  const component = renderer.create(<Column id='colId' className='colClassname' style='colStyle'>
      <div className='colChildren'></div>
    </Column>)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

it('Column renders the children', () => {
  render(<Column id='colId' className={'colClassname'} style={{ display: 'block' }}>
          <div className='colChildren'>Column Content</div>
      </Column>, container)

  expect(screen.getByText('Column Content')).toBeTruthy()
})
