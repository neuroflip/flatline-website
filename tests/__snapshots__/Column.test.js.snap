// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column renders as expected 1`] = `
<div
  className="colClassname"
  id="colId"
  style="colStyle"
>
  <div
    className="colChildren"
  />
</div>
`;
